datalad/interface/run_procedure.py

```diff
@@ -47,26 +49,64 @@ def _get_file_match(dir, name='*'):
     for target in targets:
         for m in iglob(op.join(dir, target)):
             m_bn = op.basename(m)
-            if name == '*' or m_bn == name or m_bn.startswith('{}.'.format(name)):
-                yield m
+            if name == '*':
+                report_name = m_bn[:-3] if m_bn.endswith('.py') or \
+                                           m_bn.endswith('.sh') \
+                                        else m_bn
+                yield m, report_name
+            elif m_bn == name or m_bn.startswith('{}.'.format(name)):
+                yield m, name
+
+
+def _get_proc_config(name, ds=None):
+    """get configuration of named procedure
+
+    Figures call format string and help message for a given procedure name,
+    based on dataset.
+
+    Returns
+    -------
+    tuple
+      (call format string, help string) or possibly None for either value,
+      if there's nothing configured
+    """
+    # figure what ConfigManager to ask
+    cm = cfg if ds is None else ds.config
+    v = cm.get('datalad.procedures.{}.call-format'.format(name), None)
+    h = cm.get('datalad.procedures.{}.help'.format(name), None)
+    if isinstance(v, tuple):
+        # If multiple values are defined (at the same level), ConfigManager
+        # returns a tuple. Go with the last one.
+        # TODO: Should we issue a warning or fail entirely?
+        return v[-1], h
+    else:
+        return v, h
 
 
 def _get_procedure_implementation(name='*', ds=None):
     ds = ds if isinstance(ds, Dataset) else Dataset(ds) if ds else None
-    # 1. check dataset for procedure
+
+    # 1. check system and user account for procedure
```

Comment:
I don't have a strong preference on what the precedence should be, but since you're changing the original order, it would have been nice if the commit message gave some rational for this.

And I think the order in "Directories identified by the configuration settings" of the help is supposed to follow the order, so that should be updated.

Reply:
You're right, of course.